Update EBNF for password options (#18090)
pingcap/docs: sql-statements/sql-statement-create-user.md

The following `CREATE USER` options are not yet supported by TiDB, and will be parsed but ignored:

* `PASSWORD REQUIRE CURRENT DEFAULT`
* `WITH MAX_QUERIES_PER_HOUR`
* `WITH MAX_UPDATES_PER_HOUR`
* `WITH MAX_USER_CONNECTIONS`

> The following `CREATE USER` options are also not supported by TiDB, and are *not* accepted by the parser:

Reviewer: ```suggestion
The following `CREATE USER` options are not supported by TiDB either, and are *not* accepted by the parser:
```

> The following `CREATE USER` options are not yet supported by TiDB, and will be parsed but ignored:
> 
> * TiDB does not support `WITH MAX_QUERIES_PER_HOUR`, `WITH MAX_UPDATES_PER_HOUR`, and `WITH MAX_USER_CONNECTIONS` options.
> * TiDB does not support `PASSWORD REQUIRE CURRENT DEFAULT`.

Reviewer 2: ```suggestion
The following `CREATE USER` options are not yet supported by TiDB, and will be parsed as ignored by the parser:

* `WITH MAX_QUERIES_PER_HOUR`
* `WITH MAX_UPDATES_PER_HOUR`
* `WITH MAX_USER_CONNECTIONS`
* `PASSWORD REQUIRE CURRENT DEFAULT`
```

> * TiDB does not support `WITH MAX_QUERIES_PER_HOUR`, `WITH MAX_UPDATES_PER_HOUR`, and `WITH MAX_USER_CONNECTIONS` options.
> * TiDB does not support `PASSWORD REQUIRE CURRENT DEFAULT`.

Reviewer 2: ```suggestion
* `WITH MAX_QUERIES_PER_HOUR`
* `WITH MAX_UPDATES_PER_HOUR`
* `WITH MAX_USER_CONNECTIONS`
* `PASSWORD REQUIRE CURRENT DEFAULT`
```

> In addition to that these options are also not supported by TiDB, and are *not* accepted by the parser:
> 
> * TiDB does not support the `DEFAULT ROLE` option.
> * TiDB does not support `PASSWORD REQUIRE CURRENT OPTIONAL`.

Reviewer 2: ```suggestion
The following `CREATE USER` options are also not supported by TiDB, and are *not* accepted by the parser:

* `DEFAULT ROLE`
* `PASSWORD REQUIRE CURRENT OPTIONAL`
```
